pkg/planner/core/memtable_predicate_extractor.go

```diff
@@ -88,72 +86,132 @@ func (extractHelper) extractColInConsExpr(ctx base.PlanContext, extractCols map[
 	return name.ColName.L, results
 }
 
-func (helper *extractHelper) extractColBinaryOpConsExpr(ctx base.PlanContext, extractCols map[int64]*types.FieldName, supportLower bool, expr *expression.ScalarFunction) (string, []types.Datum) {
+func (helper *extractHelper) setColumnPushedDownFn(
+	colNameL string,
+	extractCols map[int64]*types.FieldName,
+	expr *expression.ScalarFunction,
+) {
+	scalar := helper.extractColBinaryOpScalarFunc(extractCols, expr)
+	if scalar == nil {
+		return
+	}
+	switch scalar.FuncName.L {
+	case ast.Lower:
+		helper.pushedDownFuncs = make(map[string]func(string) string)
```

Comment:
`pushedDownFuncs` is created each time `setColumnPushedDownFn` called?

Reply:
I reduce the number of times to call `setColumnPushedDownFn` in [this commit](https://github.com/pingcap/tidb/pull/55844/commits/be4f3def6504c1ce01387b187dbc4bba336fe056)